### PR TITLE
update webpack loader options

### PR DESCRIPTION
**Pull request by smackgg**
Add resourceQuery param.

```js
import styles from './styles.css?type=global'
// webpack
config: {
  module: {
    rules: [
      test: /\.css$/,
      use: [{
        loader: require('styled-jsx/webpack').loader,
        options: {
          type: (fileName, fileNameQuery) => fileNameQuery.type || 'scoped'
        }
      }]
    ]
  }
}
```

It is necessary to use resourceQuery sometimes.

Such as https://github.com/zeit/styled-jsx/issues/521

### CODE REVIEW

File: src/webpack.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function(content) {
 
   // Calls type with the current file name.
   if (typeof options.type === 'function') {
-    options.type = options.type(this.resourcePath)
+    options.type = options.type(this.resourcePath, loaderUtils.parseQuery(this.resourceQuery || '?'))
```
Comment (giuseppeg) [inline src/webpack.js]: can you make the second argument an object with a `resourceQuery` key? This way we can add more options if necessary

Comment (smackgg) [reply src/webpack.js]: yep

File: readme.md [hunk 2]
```diff
@@ -593,15 +593,33 @@ config: {
...
+
+// webpack
+config: {
+  module: {
+    rules: [
+      test: /\.css$/,
+      use: [{
+        loader: require('styled-jsx/webpack').loader,
+        options: {
+          type: (fileName, fileNameQuery) => fileNameQuery.type || 'scoped'
```
Comment (giuseppeg) [inline readme.md]:
```suggestion
          type: (fileName, options) => options.query.type || 'scoped'
```

File: src/webpack.js [hunk 3]
```diff
@@ -13,7 +13,9 @@ module.exports = function(content) {
 
   // Calls type with the current file name.
   if (typeof options.type === 'function') {
-    options.type = options.type(this.resourcePath)
+    options.type = options.type(this.resourcePath, {
+      parseQuery: loaderUtils.parseQuery(this.resourceQuery || '?')
```
Comment (giuseppeg) [inline src/webpack.js]:
```suggestion
      query: loaderUtils.parseQuery(this.resourceQuery || '?') || {}
```

**Comment by giuseppeg** [comment]
@smackgg I am terribly sorry for the delayed reaction. If you commit my suggestions I will merge right away.

**Comment by smackgg** [comment]
@giuseppeg ok

**Pull request merged**